src/software/ai/hl/stp/skill/shoot/shoot_skill_fsm.cpp

```diff
@@ -1,19 +1,60 @@
 #include "software/ai/hl/stp/skill/shoot/shoot_skill_fsm.h"
 
-void ShootSkillFSM::pivotKick(
-    const Update& event,
-    boost::sml::back::process<PivotKickSkillFSM::Update> processEvent)
+void ShootSkillFSM::GetPossessionFSM::getPossession(
+    const Update& event, boost::sml::back::process<DribbleSkillFSM::Update> processEvent)
 {
     Point ball_position = event.common.world_ptr->ball().position();
-    Point kick_target   = event.common.world_ptr->field().enemyGoalCenter();
 
-    std::optional<Shot> best_shot =
-        (*event.common.strategy)->getBestShot(event.common.robot);
-    if (best_shot)
+    DribbleSkillFSM::ControlParams control_params{
+        .dribble_destination = ball_position,
+        .final_dribble_orientation =
+            (event.common.world_ptr->field().enemyGoalCenter() - ball_position)
+                .orientation(),
+        .allow_excessive_dribbling = false};
+
+    processEvent(DribbleSkillFSM::Update(control_params, event.common));
+}
+
+void ShootSkillFSM::getPossession(
+    const Update& event, boost::sml::back::process<GetPossessionFSM::Update> processEvent)
+{
+    processEvent(GetPossessionFSM::Update({}, event.common));
+}
+
+void ShootSkillFSM::dribbleBallToKickOrigin(
+    const Update& event, boost::sml::back::process<DribbleSkillFSM::Update> processEvent)
+{
+    if (!best_shot_)
     {
-        kick_target = best_shot->getPointToShootAt();
+        best_shot_ = (*event.common.strategy)->getBestShot(event.common.robot);
+
+        if (!best_shot_)
+        {
+            // Default kick origin and target if no shot found
+            Point kick_origin = event.common.world_ptr->ball().position();
+            Point kick_target = event.common.world_ptr->field().enemyGoalCenter();
+            best_shot_        = Shot(kick_origin, kick_target, Angle::zero());
```

Comment:
suggestion: try a chip shot or do a 50/50 coin-flip for kick/chip to confuse enemy team.

Reply:
suggestion: maybe add a `MAX_SHOT_SEARCH_ATTEMPTS` and set to 3. then add a small delay interval to potentially look for a shot and wait before trying again.

Reply:
We'll have a `ChipSkill`, so the SkillGraph will dictate whether we kick or chip